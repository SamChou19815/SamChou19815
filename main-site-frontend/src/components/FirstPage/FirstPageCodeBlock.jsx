// @flow strict

import type { Node } from 'react';
import React from 'react';
import CodeBlock from '../Common/CodeBlock';
import styles from './FirstPageCodeBlock.module.css';

const code = `

/**
 * Copyright (C) 2015â€“2019 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com
 
class List<T>(Nil(unit), Cons([T * List<T>])) {
  public function <T> of(t: T): List<T> =
    Cons([t, Nil(unit)])
  public method cons(t: T): List<T> =
    Cons([t, this])
}
class Developer(
  name: string, github: string,
  projects: List<string>,
) {
  public function sam(): Developer =
    val l = List::of("SAMLANG")::cons("...");
    val github = "SamChou19815";
    { name: "Sam Zhou", github, projects: l }
}
util Main {
  function main(): Developer = Developer::sam()
}
`;

export default (): Node => <CodeBlock className={styles.Block}>{code}</CodeBlock>;
