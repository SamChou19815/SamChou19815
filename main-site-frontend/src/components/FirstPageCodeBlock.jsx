// @flow strict

import type { Node } from 'react';
import React from 'react';
import CodeBlock from './CodeBlock';
import styles from './FirstPageCodeBlock.module.css';

const code = `
/**
 * © 2015–2019 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com
 
class Birthday(year: int, month: int, day: int) {
  public function ofSam(): Birthday =
    { year: 1998, month: 11, day: 15 }
}
class Developer(
  name: string, university: string,
  github: string, birthday: Birthday,
) {
  public function sam(): Developer =
    val university = "Cornell University";
    val github = "SamChou19815";
    { 
      name: "Sam Zhou", university, github,
      birthday: Birthday::ofSam(),
    }
}
util Main {
  function main(): Developer = Developer::sam()
}
`;

export default (): Node => <CodeBlock className={styles.Block}>{code}</CodeBlock>;
