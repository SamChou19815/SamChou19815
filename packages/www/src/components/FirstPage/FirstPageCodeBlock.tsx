import React, { ReactElement } from 'react';

import CodeBlock from 'lib-react/PrismCodeBlock';

import styles from './FirstPageCodeBlock.module.css';

const code = `

/**
 * Copyright (C) 2015â€“2020 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com

class List<T>(Nil(unit), Cons([T * List<T>])) {
  function <T> of(t: T): List<T> =
    Cons([t, Nil({})])
  method cons(t: T): List<T> =
    Cons([t, this])
}
class Developer(
  val name: string, val github: string,
  val projects: List<string>,
) {
  function sam(): Developer = {
    val l = List.of("SAMLANG").cons("...")
    val github = "SamChou19815"
    { name: "Sam Zhou", github, projects: l }
  }
}
class Main {
  function main(): Developer = Developer.sam()
}
`;

export default (): ReactElement => (
  <CodeBlock language="samlang" className={styles.Block}>
    {code}
  </CodeBlock>
);
