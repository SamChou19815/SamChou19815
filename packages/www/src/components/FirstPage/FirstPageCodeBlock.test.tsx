import React from 'react';

import renderer from 'react-test-renderer';

import FirstPageCodeBlock from './FirstPageCodeBlock';

it('FirstPageCodeBlock matches snapshot.', () => {
  const tree = renderer.create(<FirstPageCodeBlock />).toJSON();
  expect(tree).toMatchSnapshot();
});
