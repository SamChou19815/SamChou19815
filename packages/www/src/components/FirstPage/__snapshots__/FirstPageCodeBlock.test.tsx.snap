// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstPageCodeBlock matches snapshot. 1`] = `
<pre
  className="Block"
>
  <code
    className="samlang"
  >
    

/**
 * Copyright (C) 2015â€“2020 Developer Sam.
 *
 * @author sam
 */

// Basic information of sam, written in SAMLANG.
// The code below is a well-typed SAMLANG program.
// Try it by yourself at
// https://samlang-demo.developersam.com

class List&lt;T&gt;(Nil(unit), Cons([T * List&lt;T&gt;])) {
  public function &lt;T&gt; of(t: T): List&lt;T&gt; =
    Cons([t, Nil(unit)])
  public method cons(t: T): List&lt;T&gt; =
    Cons([t, this])
}
class Developer(
  val name: string, val github: string,
  val projects: List&lt;string&gt;,
) {
  public function sam(): Developer = {
    val l = List.of("SAMLANG").cons("...")
    val github = "SamChou19815"
    { name: "Sam Zhou", github, projects: l }
  }
}
class Main {
  function main(): Developer = Developer.sam()
}

  </code>
</pre>
`;
